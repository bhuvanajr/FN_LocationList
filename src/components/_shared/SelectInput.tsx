import { ErrorMessage } from '@hookform/error-message';
import React from 'react';
import { useFormContext } from 'react-hook-form';
import { ErrorMessageDisplay } from './ErrorMessageDisplay';

type Props = {
  name: string;
  id?: string;
  options: string[];
  placeholder: string;
  rules?: Object;
};

export const SelectInput = (props: Props) => {
  const {
    register,
    formState: { errors },
  } = useFormContext();

  return (
    <>
      <select
        {...register(props.name, props.rules)}
        name={props.name}
        id={props.id}
        className="form-control form-select"
      >
        <option value="">{props.placeholder}</option>
        {props.options.map((option, index) => (
          <option value={option} key={index}>
            {option}
          </option>
        ))}
      </select>
      {props.rules && (
        <ErrorMessage
          errors={errors}
          name={props.name}
          render={({ message }): React.ReactNode => <ErrorMessageDisplay id={props.name} message={message} />}
        />
      )}
    </>
  );
};
