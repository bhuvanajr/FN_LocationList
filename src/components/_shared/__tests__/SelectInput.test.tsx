import { render, screen } from '@testing-library/react';
import React from 'react';
import { SelectInput } from '../SelectInput';

jest.mock('../ErrorMessageDisplay.tsx', () => ({
  ErrorMessageDisplay: () => <div>ErrorMessageDisplay</div>,
}));

jest.mock('react-hook-form', () => ({
  ...jest.requireActual('react-hook-form'),
  useFormContext: () => ({
    register: () => jest.fn(),
    formState: () => jest.fn(),
  }),
}));

jest.mock('@hookform/error-message', () => ({
  ErrorMessage: () => <div>ErrorMessage</div>,
}));

test('should render SelectInput correctly', () => {
  const { container } = render(
    <SelectInput name="mockname" placeholder="choose input" options={['option1', 'option2']} />
  );
  expect(container).toMatchSnapshot();
});

test('should render SelectInput with ErrorMessage correctly when rules are passed', () => {
  render(
    <SelectInput
      name="mockname"
      placeholder="choose input"
      options={['option1', 'option2']}
      rules={{ required: true }}
    />
  );

  const errorMessage = screen.getByText(/ErrorMessage/i);
  expect(errorMessage).toBeInTheDocument();
});
