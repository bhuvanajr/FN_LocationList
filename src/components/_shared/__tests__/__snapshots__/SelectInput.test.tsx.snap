// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SelectInput correctly 1`] = `
<div>
  <select
    class="form-control form-select"
    mock="[object Object]"
    name="mockname"
  >
    <option
      value=""
    >
      choose input
    </option>
    <option
      value="option1"
    >
      option1
    </option>
    <option
      value="option2"
    >
      option2
    </option>
  </select>
</div>
`;
